packages/istanbul-lib-instrument/src/visitor.js

```diff
@@ -154,7 +154,9 @@ class VisitState {
             // make an attempt to hoist the statement counter, so that
             // function names are maintained.
             const parent = path.parentPath.parentPath;
-            if (parent && (T.isProgram(parent.parentPath) || T.isBlockStatement(parent.parentPath))) {
+            if (parent && T.isExportNamedDeclaration(parent.parentPath)) {
```

Comment:
I really like how minimal this change is, it's great that we're targeting just the `ExportNamedDeclaration` type.